src/MetaMorpho.sol

```diff
@@ -673,11 +673,10 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
 
     /// @inheritdoc ERC4626
     /// @dev Used in redeem or withdraw to withdraw the underlying asset from Morpho markets.
-    /// @dev Depending on 4 cases, reverts when withdrawing "too much" with:
+    /// @dev Depending on 3 cases, reverts when withdrawing "too much" with:
```

Comment:
```suggestion
    /// @dev Depending on 3 cases, reverts when withdrawing "too much" with:
    /// 1. NotEnoughLiquidity when withdrawing more than available liquidity.
    /// 2. ERC20InsufficientAllowance when withdrawing more than `caller`'s allowance.
    /// 3. ERC20InsufficientBalance when withdrawing more than `owner`'s balance.
```
- put `NotEnoughLiquidity` first because `_withdrawMorpho` is called before the transfer
- remove mention of total assets because it does not appear in the functions called